hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestClientSideRegionScanner.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.io.hfile.BlockCache;
+import org.apache.hadoop.hbase.io.hfile.LruBlockCache;
+import org.apache.hadoop.hbase.testclassification.ClientTests;
+import org.apache.hadoop.hbase.testclassification.SmallTests;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category({ SmallTests.class, ClientTests.class })
+public class TestClientSideRegionScanner {
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestClientSideRegionScanner.class);
+
+  private final static HBaseTestingUtil TEST_UTIL = new HBaseTestingUtil();
+
+  private Configuration conf;
+  private Path rootDir;
+  private FileSystem fs;
+  private TableDescriptor htd;
+  private RegionInfo hri;
+  private Scan scan;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    TEST_UTIL.startMiniCluster(1);
+  }
+
+  @AfterClass
+  public static void tearDownAfterClass() throws Exception {
+    TEST_UTIL.shutdownMiniCluster();
+  }
+
+  @Before
+  public void setup() throws IOException {
+    conf = TEST_UTIL.getConfiguration();
+    rootDir = TEST_UTIL.getDefaultRootDirPath();
+    fs = TEST_UTIL.getTestFileSystem();
+    htd = TEST_UTIL.getAdmin().getDescriptor(TableName.META_TABLE_NAME);
+    hri = TEST_UTIL.getAdmin().getRegions(TableName.META_TABLE_NAME).get(0);
+    scan = new Scan();
+  }
+
+  @Test
+  public void testDefaultBlockCache() throws IOException {
+    Configuration copyConf = new Configuration(conf);
+    ClientSideRegionScanner clientSideRegionScanner =
+      new ClientSideRegionScanner(copyConf, fs, rootDir, htd, hri, scan, null);
+
+    BlockCache blockCache = clientSideRegionScanner.getRegion().getBlockCache();
+    assertNotNull(blockCache);
+    assertTrue(blockCache instanceof LruBlockCache);
+
+    float actualBlockCacheRatio = copyConf
+      .getFloat(HConstants.HFILE_BLOCK_CACHE_SIZE_KEY,
+        copyConf.getFloat(HConstants.HBASE_CLIENT_SCANNER_BLOCK_CACHE_SIZE_KEY,
+          HConstants.HBASE_CLIENT_SCANNER_BLOCK_CACHE_SIZE_DEFAULT));
+
+    assertTrue(HConstants.HBASE_CLIENT_SCANNER_BLOCK_CACHE_SIZE_DEFAULT == actualBlockCacheRatio);
+  }
+
+  @Test
+  public void testConfiguredBlockCache() throws IOException {
+    Configuration copyConf = new Configuration(conf);
+    float blockCacheRatio = 0.05f;
+    copyConf.setFloat(HConstants.HBASE_CLIENT_SCANNER_BLOCK_CACHE_SIZE_KEY, blockCacheRatio);
+    ClientSideRegionScanner clientSideRegionScanner =
+      new ClientSideRegionScanner(copyConf, fs, rootDir, htd, hri, scan, null);
+
+    BlockCache blockCache = clientSideRegionScanner.getRegion().getBlockCache();
+    assertNotNull(blockCache);
+    assertTrue(blockCache instanceof LruBlockCache);
```

Comment:
Related to the above comment, LruBlockCache here will cache all blocks from the HFile, not just the index blocks which was the goal.